### ISSUE TITLE
Forwarding to splunk HEC?

### ISSUE THREAD

**Issue by aegershman** [cloudfoundry/cf-drain-cli#13]
# tl;dr
Is it possible to set up a drain which can shuttle logs via https to a splunk HEC (http endpoint collector)?

## context
Splunk HEC's require an authorization token to be supplied either in the request body or as a header when the client is sending logs. I don't think it's currently possible to supply any sort of auth token when creating a drain which the drain would include as a header. 

I know I can use the firehose nozzle to forward logs, but that requires the platform operator to set up a nozzle, and the nozzle would shuttle ALL firehose logs on to an HEC. What I'd really like is a more self-service model, where platform users create their own user-provided service / drain and have it shuttle their logs to wherever they want. 

(I apologize- this is a pretty open-ended question, so if there's no response on it after a while, I'll close it out)

**Comment by bradylove** [comment]
@aegershman We do not currently support forwarding logs to Splunk HEC, however we have been working on apps to fulfill the self-service model. Currently the protocols we are targeting for these self service syslog forwarders are the same as the cf-syslog-drain-release (RFC5424 over syslog://, syslog-tls://, and https://). These self-service apps are going to depend on a new feature of Loggregator that exposes our V2 API to the outside world via HTTP.

Without having much context on the Splunk HEC, I do have some questions that maybe you can help with:
1. Does the authentication token you would use for this ever expire or need renewing? Or is it valid until an admin or someone revokes it?
1. Does the Splunk HEC endpoint support, or have a way to configure support for syslog messages in RFC5424 format? Looking at [this document](http://docs.splunk.com/Documentation/Splunk/latest/Data/AboutHEC#How_HTTP_Event_Collector_works) it looks like it expects a JSON payload.

@ahevenor do you have any input on this?

**Comment by aegershman** [comment]
Apologies for the late reply, even though I completely agree this issue should be closed let me just follow up with these comments:

1. The auth token does not expire or need renewing. It is valid until an admin revokes it.
2. The HEC receiver is designed to listen on a port for HTTP(s) traffic. 

The formatting of the message payload is secondary to it being received. So once the message is received, it is _then_ parsed and tokenized. Splunk *can* be configured to receive on TCP and UDP ports, though if it would be pretty cool if I could do this over HTTP(s). 

Regardless, thanks a ton for your time. I appreciate it. 👍
